vault/dynamic_system_view.go

```diff
@@ -145,3 +145,33 @@ func (d dynamicSystemView) LookupPlugin(ctx context.Context, name string) (*plug
 func (d dynamicSystemView) MlockEnabled() bool {
 	return d.core.enableMlock
 }
+
+func (d dynamicSystemView) EntityInfo(entityID string) (*logical.Entity, error) {
+	if d.core == nil {
+		return nil, fmt.Errorf("system view core is nil")
+	}
+	if d.core.identityStore == nil {
+		return nil, fmt.Errorf("system view identity store is nil")
+	}
+	// Get a clone of the current entity id in the store
+	entity, err := d.core.identityStore.MemDBEntityByID(entityID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	aliases := make([]*logical.Alias, len(entity.Aliases))
+	for i, alias := range entity.Aliases {
+		aliases[i] = &logical.Alias{
+			MountType:     alias.MountType,
```

Comment:
Aliases do not hold mount type, this value will always be blank from a pure lookup. See https://github.com/hashicorp/vault/blob/master/vault/identity_store_aliases.go#L451 for how to look it up in real time.

In theory for aliases it _might_ be okay to store the type (just not the path). But if we have to look up the path anyways in real-time (since it can change) we are already getting the type...and we have had cases in the past where we've upgraded one type to another (e.g. aws-ec2 to aws) so it's safer.